pkg/summary/collector_test.go

```diff
@@ -0,0 +1,46 @@
+package summary
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/pingcap/check"
+	"go.uber.org/zap"
+)
+
+func TestT(t *testing.T) {
+	TestingT(t)
+}
+
+var _ = Suite(&testCollectorSuite{})
+
+type testCollectorSuite struct {
+}
+
+func (suit *testCollectorSuite) TestSumDurationInt(c *C) {
+	fields := []zap.Field{}
+	logger := func(msg string, fs ...zap.Field) {
+		fields = append(fields, fs...)
+	}
+	col := newLogCollector(logger)
+	col.CollectDuration("a", time.Second)
+	col.CollectDuration("b", time.Second)
+	col.CollectDuration("b", time.Second)
+	col.CollectInt("c", 2)
+	col.CollectInt("c", 2)
+	col.Summary("foo")
+
+	c.Assert(len(fields), Equals, 3)
+	assertContains := func(field zap.Field) {
+		for _, f := range fields {
+			if f.Key == field.Key {
+				c.Assert(f, DeepEquals, field)
+				return
+			}
+		}
+		c.Assert(fields, DeepEquals, field) // Must fail and prints detail
```

Comment:
??? use [`c.Error`](https://godoc.org/github.com/pingcap/check#C.Error) if this must fail